generator/sbpg/test_structs.py

```diff
@@ -50,13 +56,29 @@ def filepath(self):
     filepath, filename = "/".join(split[:-1]), "_".join(split[-2:])
     return (filepath, filename)
 
+  def to_json(self):
+    return {
+      'description': self.description,
+      'generated_on': self.generated_on,
+      'package': self.package,
+      'tests': [test.to_json() for test in self.tests],
+    }
+
+  def write(self, filename, exists_ok=False):
+    file_exists = os.path.exists(filename)
+    assert exists_ok or not file_exists
+    data = self.to_json()
+    try:
+      with open(filename, "w") as f:
+        yaml.dump(data, f)
+    except Exception:
+      os.remove(filename)
```

Comment:
this should only be done if the file did not already exist.  possibly even remove try by doing yaml.dumps outside of the `with`, so the only exceptions possible inside the `with` are fs related problems, which should ripple up.

if keeping the `try`, also re-raise the exception